tokio/src/executor/task/mod.rs

```diff
@@ -35,38 +35,54 @@ use std::ptr::NonNull;
 use std::{fmt, mem};
 
 /// An owned handle to the task, tracked by ref count
-pub(crate) struct Task<S: 'static> {
+pub(crate) struct Task<S: 'static, M = SendMarker> {
     raw: RawTask,
-    _p: PhantomData<S>,
+    _p: PhantomData<(S, M)>,
 }
 
-unsafe impl<S: Send + Sync + 'static> Send for Task<S> {}
+/// An owned handle to a `!Send` task, tracked by ref count.
+#[cfg(feature = "local")]
+pub(crate) type UnsendTask<S> = Task<S, UnsendMarker>;
+
+/// Marker type indicating that a `Task` was constructed from a future that
+/// implements `Send`.
+#[derive(Debug)]
+pub(crate) struct SendMarker {}
```

Comment:
Given that these marker types appear at much higher levels of abstraction, I'd prefer them to have names that are more descriptive at those layers. Something like `Sendable` and `Unsendable` perhaps?